docs/design.md

```diff
@@ -0,0 +1,194 @@
+# Design of image update automation
+
+There are three parts to the design, that operate independently, and
+in sympathy with one another:
+
+ - the job runner (type UpdateJob)
+ - the image metadata reflector (types ImageRepository, ImagePolicy)
+ - the automation controller (type ImageAutomation)
+
+Some tooling comes alongside these parts:
+
+ - the `tk-image` command-line tool lets you create the resource
+   mentioned above, and can be used as an extension to the gitops
+   toolkit command-line tool.
+ - the `image-update` image can be used with `kpt fn`, GitHub Actions,
+   and `UpdateJob` to update images in resources within a working
+   directory.
+
+## Update job controller
+
+The job controller runs `UpdateJob` resources, each of which specifies
+an update operation on a git repository. Each run checks out the given
+repo at the specified ref, runs the specified image with the
+arguments, then commits and pushes as specified.
```

Comment:
I think we could highlight that the source branch/tag can be different to the destination branch. One workflow that I have in mind is:
- fetch manifests from the latest semver tag 
- commit/push patch to the staging branch
- GH Action opens a PR from staging into master
- cluster admin merges the PR and create a GH release
- source-controller pulls the new semver tag
- kustomize/helm reconcilers apply the changes on the cluster